kernel/dev/rtc/rtc.cpp

```diff
@@ -136,15 +139,34 @@ void read_rtc() {
     }
 
     // Calculate the full (4-digit) year
-    if(RTC_CURRENT_CENTURY != 0) {
+    if (RTC_CURRENT_CENTURY != 0) {
         rtc_year += rtc_century * 100;
     } else {
         rtc_year += (RTC_CURRENT_YEAR / 100) * 100;
-        if(rtc_year < RTC_CURRENT_YEAR) rtc_year += 100;
+        if (rtc_year < RTC_CURRENT_YEAR)
+            rtc_year += 100;
     }
 }
+//
+// Julian date calculation from https://en.wikipedia.org/wiki/Julian_day
+static uint64_t get_jdn(uint8_t days, uint8_t months, uint16_t years)
```

Comment:
I know you've only just started working on this, but I wanted to say that it's generally bad form to just copy and paste code from somewhere else. We don't want to copy the code out of Limine but instead use it as a guide and inspiration for our own project, even if that means that the result is different only in style (which would likely be the case here since this is basically just math).

For example, the Panix way of doing things would be to rename the functions as `getJulianDay` and `getUnixEpoch`. Same thing for the variables. `jdnNow` and `jdn1970` are both valid options.

Hope that makes sense. 🙂

Reply:
oh, thanks for the tip, i'll rework in this code later.